app/routines/user_from_signup_state.rb

```diff
@@ -6,7 +6,7 @@ class UserFromSignupState
 
   def exec(signup_state)
     user = User.new
-    if signup_state.trusted?
+    if signup_state && signup_state.trusted?
       user.role = signup_state.role
```

Comment:
These work and are arguably the way to go.  There is a rubyism that uses `try` to work around possibly nil values. 
For instance this could be:
```
signup_state.try(:trusted?)
```
Personally I find the <value> && <value>.<method> equally clear though.